### PR TITLE
fabtests/prov/lpp: change invalid condition in fi_tsenddata

### PR DESCRIPTION
**Pull request by tstruk**
Change invalid condition in parameter check in fi_tsenddata test.

### CODE REVIEW

File: fabtests/prov/lpp/src/rcq_data.c [hunk 1]
```diff
@@ -16,7 +16,7 @@ int run_fi_tsenddata(struct rank_info *ri){
 	size_t ndata = sizeof(rcq_data)/sizeof(*rcq_data);
 	size_t nbufflens = sizeof(buff_lens)/sizeof(*buff_lens);
 
-	if (ndata == nbufflens)
+	if (ndata != nbufflens)
 		return -EINVAL;
```
Comment (j-xiong) [inline fabtests/prov/lpp/src/rcq_data.c:20]: Why having this check at all? The two values are always equal based on how the two arrays are initialized.

Comment (tstruk) [reply fabtests/prov/lpp/src/rcq_data.c:20]: That was an assert probably left there as the code evolved.
I have reworked the code to use buffers with static size and removed to check.

Comment (tstruk) [reply fabtests/prov/lpp/src/rcq_data.c]: PR updated.

**Comment by j-xiong** [comment]
@tstruk Please rebase to resolve conflicts.

**Review by j-xiong — APPROVED** [review]

**Pull request merged**